### DISCUSSION TITLE
7 ERROR: relation "deferred_bindings" does not exist

### DISCUSSION THREAD

**Discussion by TrenchTown** [wintercms/winter#738] [Q&A] [answered]
I use 2 Mysql and Postgresql databases when I try to save a record I get an error
Winter version 1.2.1

"SQLSTATE[42P01]: Undefined table: 7 ERROR: relation "deferred_bindings" does not exist
LINE 1: select * from "deferred_bindings" where "master_type" = $1 a...
^ (SQL: select * from "deferred_bindings" where "master_type" = Pixellife\Admin\Models\Fgo and "session_key" = MLpDACy2KLl1EnBi0DiPQvHCkKWiSIsVfgj3WHv3)" on line 760 of C:\OSPanel\domains\admin.local\vendor\laravel\framework\src\Illuminate\Database\Connection.php

**Comment by LukeTowers** [comment, marked as answer]
What do you mean you use 2 databases? Do they both have the deferred bindings tables setup?

**Reply by TrenchTown** [reply]
> What do you mean you use 2 databases? Do they both have the deferred bindings tables setup?

In the file database.php specified the connection settings for mysql and pgsql databases. I see my data in the administration panel
from pgsql, but when I try to save this error pops up.

"Do they both have the deferred bindings tables setup?" - I do not know how to do this, please show an example

**Reply by LukeTowers** [reply]
It sounds like you have a model that is using a database connection that is not the default connection that Winter would have initialized on (and thus created the deferred_bindings table on for you). You can either create the deferred_bindings table in your model's connection yourself, or you can disable the deferred binding functionality entirely on the models that use the secondary connection: https://wintercms.com/docs/database/relations#disable-deferred-binding

**Comment by TrenchTown** [comment]
I added a function to the model and everything worked, but I'm not sure about the correctness of such a decision, how much is it correct?
```
public function save(?array $options = [], $sessionKey = null)
{
    $this->sessionKey = null;
    return $this->saveInternal(['force' => false] + (array) $options);
}
```

**Reply by LukeTowers** [reply]
It is recommended to use the approach I linked in the documentation previously.

**Reply by TrenchTown** [reply]
I don't understand how to do this. There is not a single example in the whole Internet, could you give an example of what I should prescribe in my model?

**Reply by LukeTowers** [reply]
See https://wintercms.com/docs/database/relations#disable-deferred-binding.

<img width="996" alt="Screen Shot 2022-10-26 at 2 05 43 PM" src="https://user-images.githubusercontent.com/7253840/198126042-f6f151df-0cb0-418b-b01f-5bc02483a707.png">

```php
public function __construct()
{
    $result = parent::__construct(...func_get_args());
    $this->bindEvent('model.saveInternal', function () {
        $this->sessionKey = null;
    });
    return $result;
}
```

**Reply by TrenchTown** [reply]
It doesn't work that way, it gives an error
"Declaration of Toman\Admin\Models\Fgo::__construct() must be compatible with Winter\Storm\Database\ModelInterface::__construct(array $attributes = [])"

```
public function __construct()
{
    $result = parent::__construct(...func_get_args());
    $this->bindEvent('model.saveInternal', function () {
        $this->sessionKey = null;
    });
    return $result;
}
```
It worked like this
```
public function __construct(array $attributes = [])
{
        $result = parent::__construct(...func_get_args());

        $this->bindEvent('model.saveInternal', function () {
            $this->sessionKey = null;
        });
        return $result;
}
```

**Reply by LukeTowers** [reply]
Fixed in https://github.com/wintercms/docs/commit/8fbd046151ad93e7146dfd8f1a885e29980e8a71, thanks!
